src/Tracking/Events/Activated.php

```diff
@@ -0,0 +1,65 @@
+<?php
+declare( strict_types=1 );
+
+namespace Automattic\WooCommerce\GoogleListingsAndAds\Tracking\Events;
+
+/**
+ * This class adds actions to track when the extension is activated.
+ * *
+ *
+ * @package Automattic\WooCommerce\GoogleListingsAndAds\Tracking
+ */
+class Activated extends BaseEvent {
+
+	/**
+	 * The page where activation with a source can occur.
+	 */
+	public const ACTIVATION_PAGE = 'plugin-install.php';
+
+	/**
+	 * The query parameters used to determine activation source details.
+	 */
+	public const SOURCE_PARAMS = [
+		'traffic_source',
+		'traffic_type',
+	];
+
+	/**
+	 * Register the tracking class.
+	 */
+	public function register(): void {
+		add_action( 'woocommerce_gla_extension_activated', [ $this, 'maybe_track_activation_source' ] );
```

Comment:
I can't find a `do_action` for this one. Is this something that gets called dynamically? It seems that `maybe_track_activation_source` is manually called in the `Infrastructure\GoogleListingsAndAdsPlugin::activate` method.

Reply:
Yeah, only manually invoked now. I had this one in there in case someone found a way to do this with actions, but I'm convinced it's not possible. Removed in 5ea87a0cf12e10b0903d922987710621e8206509